wled00/FX_fcn.cpp

```diff
@@ -438,18 +438,21 @@ uint32_t IRAM_ATTR Segment::currentColor(uint8_t slot) {
 #endif
 }
 
-CRGBPalette16 IRAM_ATTR &Segment::currentPalette(CRGBPalette16 &targetPalette, uint8_t pal) {
-  loadPalette(targetPalette, pal);
-  unsigned prog = progress();
+inline CRGBPalette16 &Segment::currentPalette(uint8_t pal) {
+  if ( SEGPALETTEIDX != pal ) {
```

Comment:
This will not work during mode (aka effect) blending. Each mode (effect) can have its own palette that needs to be blended during execution of mode (effect).